### PR TITLE
-e parameter now handle regular espression

### PR DESCRIPTION
**Pull request by wallaroo**
if only one -e parameter is passed to cli and the value matches /.*/[gmiuy]{0,5} then this is treated as a Regular Expression

### CODE REVIEW

**Comment by alexyork** [comment]
Today I discovered I need this feature, also. Can someone on the `react-docgen` team merge this into master and ship a new version?

My usecase is this: I want to point to a `/components` folder and create docs for my components but exclude all files that match the pattern `*.example.js`. Currently `-e *.example.js` sends in `['*.example.js']` as a string, not a RegEx, so I cannot achieve what I want via the CLI.

I tried this PR and passed in `-e /.example.js/` and it worked as expected.

I would love it if this got merged in :-)

**Comment by fkling** [comment]
I will look at it over the weekend!

File: bin/react-docgen.js [hunk 1]
```diff
@@ -68,6 +68,11 @@ var ignoreDir = argv.ignore;
 var excludePatterns = argv.exclude;
 var resolver;
 var errorMessage;
+var regexRegex = /\/(.*)\/([igymu]{0,5})/;
```
Comment (fkling) [inline bin/react-docgen.js]: The pattern should be anchored: 

    /^\/(.*)\/([igymu]{0,5})$/;

Otherwise it looks fine to me!

**Comment by fkling** [comment]
Merged as d9782c8a5d6750e35a8fc665bc446c02c9e027c7 . Will release a new version soon.

**Comment by fkling** [comment]
Released v2.16.0!

**Pull request closed without merging**